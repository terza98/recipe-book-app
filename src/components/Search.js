import React, { useState, useEffect } from 'react';

import Autocomplete from '@material-ui/lab/Autocomplete';
import TextField from '@material-ui/core/TextField';

//bootstrap components
import ListGroup from 'react-bootstrap/ListGroup';
import Jumbotron from 'react-bootstrap/Jumbotron';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

export default function Search(props) {
	const [isLoaded, setIsLoaded] = useState(false);
	const [value, setValue] = React.useState('');
	const [inputValue, setInputValue] = React.useState('');

	useEffect(() => {
		setIsLoaded(true);
	}, []);

	if (!isLoaded) {
		return 'Loading...';
	} else
		return (
			<>
				<Jumbotron className="banner">
					<h1>Welcome to my Recipes app!</h1>
					<Form data-testid="form" onSubmit={props.onSubmit}>
						<Autocomplete
							id="controllable-states-demo"
							options={props.predictions}
							getOptionLabel={option => option.title}
							onChange={() =>
								setTimeout(function () {
									props.selectQuery(
										document.getElementById(
											'controllable-states-demo',
										).value,
									);
								}, 200)
							}
							style={{ width: 300 }}
							renderInput={params => (
								<TextField
									{...params}
									placeholder="Type your next meal..."
									variant="outlined"
									id="query"
									value={props.query}
									onChange={props.handleInputChange}
								/>
							)}
						/>
						<Button
							data-testid="button"
							type="submit"
							id="search-btn"
						>
							Search
						</Button>
					</Form>
				</Jumbotron>
			</>
		);
}
