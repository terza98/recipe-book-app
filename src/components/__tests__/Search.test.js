import React from 'react';
import Search from '../Search';
import ReactDOM from 'react-dom';
import { render, fireEvent, getByLabelText } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';

let container = null;
beforeEach(() => {
	// setup a DOM element as a render target
	container = document.createElement('div');
	document.body.appendChild(container);
});

afterEach(() => {
	// cleanup on exiting
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Search />, div);
});

it('matches snapshot', () => {
	const { asFragment } = render(<Search />);

	expect(asFragment()).toMatchSnapshot();
});

test('should submit when clicking submit button', () => {
	const handleSubmit = jest.fn();
	const { getByTestId } = render(<Search onSubmit={handleSubmit} />);

	const button = getByTestId('button');

	fireEvent.click(button);

	expect(handleSubmit).toHaveBeenCalled();
});
