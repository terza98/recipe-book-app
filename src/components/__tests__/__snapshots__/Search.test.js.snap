// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="banner jumbotron"
  >
    <h1>
      Welcome to my Recipes app!
    </h1>
    <form
      class=""
      data-testid="form"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        role="combobox"
        style="width: 300px;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="controllable-states-demo"
            id="controllable-states-demo-label"
          >
            Type your inquiry
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="controllable-states-demo"
              placeholder="Type your next meal..."
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Type your inquiry
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        class="btn btn-primary"
        data-testid="button"
        id="search-btn"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
